### PR TITLE
Feature/arrays

### PR DESCRIPTION
**Pull request by CaliZam**

### CODE REVIEW

File: staff/daniel-zamora/arrays/array-functions/filter.spec.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+describe("filter", function () {
+  it("Should itereate on the array and return a variable with the value of the match between array and current value.", function () {
+    var array = [1, 2, 3, 4, 5];
+
+   filter(array, function())
+    
+
+    expect(result[0]).toBe(10);
+    expect(result[1]).toBe(20);
+    expect(result[2]).toBe(30);
```
Comment (Aitor1111) [inline staff/daniel-zamora/arrays/array-functions/filter.spec.js:10]: this function callback is not defined, the expect should return errors in jasmin

File: staff/daniel-zamora/arrays/array-functions/map.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// TODO
+function map(array, expression) {
+    var result = [];
+    for (var i = 0; i < array.length; i++)
+       expression(array[i],i);
```
Comment (Aitor1111) [inline staff/daniel-zamora/arrays/array-functions/map.js:5]: You can also include array in the expression like shows mdn docs of js: 
```arr.map(function callback(currentValue, index, array)```

File: staff/daniel-zamora/arrays/array-functions/filter.spec.js (hunk 1 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/filter.spec.js:10]: Right ma men, i'm on it.

File: staff/daniel-zamora/arrays/array-functions/map.js (hunk 2 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/map.js:5]: Working on it, i want to do the basic approach

File: staff/daniel-zamora/arrays/array-functions/filter.spec.js (hunk 1 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/filter.spec.js:10]: fixed :)

File: staff/daniel-zamora/arrays/array-functions/map.js (hunk 2 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/map.js:5]: Done Amigo!

File: staff/daniel-zamora/arrays/array-functions/find-index.spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+'use strict'
+describe('find-index', function () {
+<<<<<<< Updated upstream
```
[the remark this answers is not in the document]
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/find-index.spec.js]: I already fixed. Pls check and confirm if its ok

File: staff/daniel-zamora/arrays/array-functions/find-index.spec.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+describe('find-index', function () {
+    it('The findIndex() method returns the value of index of the first element in the provided array that satisfies the provided testing function', function () {
+
+        var numbers = [2, 4, 8, 12, 15, 18, 20];
+
+        var result = findIndex(numbers, function (elemento) {
+            return elemento > 10
+        })
+
+        expect(result).toBe(12);
```
[the remark this answers is not in the document]
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/find-index.spec.js:11]: Hi Alex, 

Yes, i just saw the test was duplicate from from find. I'm on it.

File: staff/daniel-zamora/arrays/arroz/arroz.prototype.every.spec.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  it("Should iterate on the array an return -1, due any match was found", function () {
+    var array = new Arroz(1, 2, 3, 4, 5, 6, 7, 8);
+
+    var matchIndex = array.every(function(number){
+        return number > 5; 
+    });
+
+    expect(matchIndex).toBeFalsy();
+  });
+
```
Comment (alexiparkhio) [inline staff/daniel-zamora/arrays/arroz/arroz.prototype.every.spec.js:21]: `Arroz.prototype.every` requires of a callback expression, you are missing the unhappy path

File: staff/daniel-zamora/arrays/arroz/arroz.prototype.filter.spec.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  it("Should iterate on the array, and don't found any match, so, should return an empty array", function () {
+    var array = new Arroz(1, 2, 3, 4, 5, 6);
+
+    var matchArray = array.filter(function(searchValue,index) {
+        return searchValue > 8;
+  },9);
+    expect(matchArray).toEqual([]);
+ 
+  });
```
Comment (alexiparkhio) [inline staff/daniel-zamora/arrays/arroz/arroz.prototype.filter.spec.js:43]: Missing unhappy path (TypeError)

File: staff/daniel-zamora/arrays/array-functions/find-index.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (CaliZam) [reply staff/daniel-zamora/arrays/array-functions/find-index.spec.js]: Hi, Alex, is already fix. Pls could you check and let me know if its ok.

File: staff/daniel-zamora/arrays/arroz/arroz.prototype.every.spec.js (hunk 5 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/arroz/arroz.prototype.every.spec.js:21]: Hi Alex, already applied. Pls check and confirm if are ok.

File: staff/daniel-zamora/arrays/arroz/arroz.prototype.filter.spec.js (hunk 6 above)
Comment (CaliZam) [reply staff/daniel-zamora/arrays/arroz/arroz.prototype.filter.spec.js:43]: hi Alex, done, pls check and let know if its ok or should i do another change

**Pull request merged**